ui/app/AppLayouts/Chat/ChatColumn/SuggestionFilter.qml

```diff
@@ -12,26 +12,33 @@ Item {
     property int lastAtPosition: 0
     property var property: ([])
 
-    Connections {
-        onFilterChanged: invalidateFilter()
-        onPropertyChanged: invalidateFilter()
-        onSourceModelChanged: invalidateFilter()
-    }
-
+    onFilterChanged: invalidateFilter()
+    onPropertyChanged: invalidateFilter()
+    onSourceModelChanged: invalidateFilter()
     Component.onCompleted: invalidateFilter()
 
     ListModel {
         id: filterModel
     }
 
     function invalidateFilter() {
-        filterModel.clear();     
+        filterModel.clear()
 
         if (!isFilteringPropertyOk())
             return
 
-        var filter = getFilter()
-        if (!isFilterReady(filter))
+        let filter = getFilter()
+        if (filter === undefined)
+            return
+
+        this.lastAtPosition = -1
```

Comment:
in this context, wouldn't be more clear to not use `this` and pass lastAtPosition as a normal parameter?

Reply:
This is used as a property from outside of this component.
Overall design of this thing is awkward, I would wish to rebuild it, but I wanted to fix the bug in the first place.

Reply:
Agree, I cross the path of this SuggestionBox too and I was quite confuse by it.